source/PlayerInfoPanel.cpp

```diff
@@ -141,6 +142,31 @@ namespace {
 			return true;
 		return lhs->RequiredCrew() < rhs->RequiredCrew();
 	}
+
+	template <InfoPanelState::ShipComparator *F>
+	bool ReverseCompare(const shared_ptr<Ship> &lhs, const shared_ptr<Ship> &rhs)
+	{
+		return F(rhs, lhs);
+	}
+
+	InfoPanelState::ShipComparator *GetReverseCompareFrom(InfoPanelState::ShipComparator *f)
+	{
+		if(f == &CompareName)
+			return &ReverseCompare<&CompareName>;
+		else if(f == &CompareModelName)
+			return &ReverseCompare<&CompareModelName>;
+		else if(f == &CompareSystem)
+			return &ReverseCompare<&CompareSystem>;
+		else if(f == &CompareShields)
+			return &ReverseCompare<&CompareShields>;
+		else if(f == &CompareHull)
+			return &ReverseCompare<&CompareHull>;
+		else if(f == &CompareFuel)
+			return &ReverseCompare<&CompareFuel>;
+		else if(f == &CompareRequiredCrew)
+			return &ReverseCompare<&CompareRequiredCrew>;
+		return nullptr;
```

Comment:
This will never happen, right?

But if we do include it maybe we should make sure a nullptr is not returned indeed? (at line 824 we use it as if it was a set for sure but it could be nullptr)
Maybe you just had to include it for the compiler to be happy, but it never actually happens (except if we receive a nullptr as a function which it seems weird to be able to do that here). If it doesn't happen, I'm not sure if we need to check for it or if we can get away with a little comment saying it?

Or just have the last `else if` be an `else` and that way we're sure to be able to receive and return a reference to a function and not a pointer to it; which is what should happen, I think.